### PR TITLE
DOC: Improved docstrings for cTrait properties

### PR DESCRIPTION
**Pull request by mdickinson**
This PR adds improved docstrings for the various cTrait properties.

### CODE REVIEW

File: traits/ctraits.c [hunk 1]
```diff
@@ -210,7 +210,7 @@ typedef struct _trait_object {
     trait_getattr getattr;           /* Get trait value handler */
     trait_setattr setattr;           /* Set trait value handler */
     trait_post_setattr post_setattr; /* Optional post 'setattr' handler */
-    PyObject *py_post_setattr;       /* Python-based post 'setattr' hndlr */
+    PyObject *py_post_setattr;       /* Python-based post 'setattr' handler */
```
Comment (mdickinson) [inline traits/ctraits.c]: Drive-by comment spelling fix, unrelated to the rest of the PR.

File: traits/ctraits.c [hunk 2]
```diff
@@ -4885,37 +4885,124 @@ static PyMethodDef trait_methods[] = {
...
+PyDoc_STRVAR(
+    ctrait_is_mapped_doc,
+    "True if this is a mapped trait, else False.\n"
+);
+
+PyDoc_STRVAR(
+    ctrait_comparison_mode_doc,
+    "Integer constant indicating when notifiers are executed.\n"
+    "\n"
+    "Valid values for this constant are 0, 1 and 2. If 0, notifiers\n"
```
Comment (corranwebster) [inline traits/ctraits.c]: There is an Enum for this in `traits.constants.ComparisonMode`.  It is probably better to refer to that for the meanings.  That way there is only one place to update should the property change.

Comment (mdickinson) [reply traits/ctraits.c]: Agreed, and done.
